src/Components/test/E2ETest/Infrastructure/WebDriverExtensions/BasicTestAppWebDriverExtensions.cs

```diff
@@ -17,7 +18,7 @@ internal static class BasicTestAppWebDriverExtensions
         public static SelectElement WaitUntilTestSelectorReady(this IWebDriver browser)
         {
             var elemToFind = By.CssSelector("#test-selector > select");
-            browser.WaitUntilExists(elemToFind, timeoutSeconds: 30, throwOnError: true);
+            browser.Exists(elemToFind);
```

Comment:
I don't follow the difference between `browser.Exists` and `browser.WaitUntilExists`, considering that both of their implementations involve waiting.

Do we need to have both of these methods? Could we drop `Exists`, since the name is harder to understand?

Reply:
This was a temporary change, I think I changed it to WaitUntilExists.

The main difference between WaitUntilExists and Exists is that the first one will throw an exception inmediately when observing errors in the BrowserLogs without waiting till the end. So this will fail faster than even a regular Assert.

Arguably we could get rid of WaitUntilExists in favor of Exists as the main value for the first one was development, which is addressed with the change that I did for allowing different configurations for dev/ci, and fail fast after first failure.

The other difference is that Exists doesn't return the found element and WaitUntilExists does, but we can fix that.

Would you be ok if we integrate everything under Exists?

Reply:
Everything here is integrated under Browser.Exists (which now returns the element)